src/tests/hotels.test.js

```diff
@@ -0,0 +1,170 @@
+import chaiHttp from 'chai-http';
+import { use, request, expect, should } from 'chai';
+import { resolve } from 'path';
+import app from '../app';
+import db from '../models';
+import Hash from '../utils/hash';
+import tokenizer from '../utils/jwt';
+import { hotelfactory, locationfactory } from './scripts/factories';
+
+should();
+use(chaiHttp);
+
+const prefix = '/api/v1';
+
+describe('/hotels', () => {
+  let token = '';
+  let requesterToken = '';
+  let notOwnerToken = '';
+  before(async () => {
+    await db.hotel.destroy({ where: {}, force: true });
+    await db.user.destroy({ where: {}, force: true });
+    await db.location.destroy({ where: {}, force: true });
+
+    await db.user.create({
+      id: 1,
+      firstName: 'John',
+      lastName: 'McCain',
+      password: Hash.generateSync('1234567e'),
+      email: 'john@mccain.com',
+      role: 'travel_administrator'
+    });
+
+    await locationfactory({
+      id: 80,
+      country: 'Rwanda',
+      city: 'kigali'
+    });
+
+    await hotelfactory({
+      id: 82,
+      locationId: 80,
+      name: 'Test hotel',
+      image: '',
+      street: 'kk 127',
+      description: 'best ever hotel',
+      services: 'service 1, service 2',
+      userId: 1
+    });
+
+    token = await tokenizer.signToken({
+      id: 1,
+      email: 'john@mccain.com',
+      isVerified: 1,
+      role: 'travel_administrator'
+    });
+
+    notOwnerToken = await tokenizer.signToken({
+      id: 2,
+      email: 'john@mccain.com',
+      isVerified: 1,
+      role: 'travel_administrator'
+    });
+
+    requesterToken = await tokenizer.signToken({
+      id: 2,
+      email: 'john@mccain.com',
+      isVerified: 1,
+      role: 'requester'
+    });
+  });
+
+  it('POST /hotels/:hotelId/rooms - should be able to add hotel rooms', (done) => {
+    request(app)
+      .post(`${prefix}/hotels/82/rooms`)
+      .set('Authorization', `Bearer ${token}`)
+      .field('name', 'Virunga')
```

Comment:
As a suggestion, don't you think the use of `set()` function instead of `field` is more readable? Like here you could do have made an object like this in the before hook:
```
let room;
room = {
  name: 'Virunga';
  description: 'Best of the best';
  type: 'VIP';
  cost: 4000
}
```
And then here you just do:
```
.set(room)
```

Reply:
I am going to try and see if it works. I am using `formdata` instead `JSON`, that's why i'm using `field`

Reply:
> I am going to try and see if it works. I am using `formdata` instead `JSON`, that's why i'm using `field`

Ah ok, that makes sense, it was just an alternative.